### ISSUE TITLE
AQL Query result changed for COLLECT used on empty data/array

### ISSUE THREAD

**Issue by CBlinguland** [arangodb/arangodb#7749]
## My Environment

* __ArangoDB Version__:        3.4.0
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Infrastructure__:               own 
* __Operating System__:        Windows 10 (same on Linux (Debian Stretch))
* __Used Package__:              Windows Installer Debian or Ubuntu .deb

## Component, Query & Data

__Affected feature__:
AQL query result differs between 3.4.0 and 3.3.20 (and lower?)

__AQL query (if applicable)__:
```aql
FOR v IN []
COLLECT collection=PARSE_IDENTIFIER(v._id) INTO grp=v
RETURN {collection: collection,grp: grp}
```

## Steps to reproduce

1. Use the AQL query
2. get result on ArangoDB 3.3.20
3. get result on ArangoDB 3.4
4. compare both results: surprise!

__Problem__:

The result for the same query executed on different version of ArangoDB Server differs.

V 3.3.20 returns:
```js
[]
```

V 3.4 returns:
```js
[
  {
    "collection": "(non-representable type)",
    "grp": []
  }
]
```

__Expected result__:

ArangoDB V 3.4 should return an empty array too (like V 3.3.20 does).
The changed query result breaks the logic in my project.

**Comment by konsultaner** [comment]
I experienced the same behavior in all my queries.

**Comment by jsteemann** [comment]
Fixed in 3.4.1.

**Comment by maxkernbach** [comment]
Hi @CBlinguland,

version 3.4.1, which includes a fix for this issue, has been released.

Could you please give it a try?

**Comment by CBlinguland** [comment]
Hi @maxkernbach,

your fix in version 3.4.1 (windows and linux) solved the problem for me.
Thanks for fixing this...
